src/renderer/utils/InstallManager.tsx

```diff
@@ -688,11 +600,192 @@ export class InstallManager {
+    const selectedTrack = store.getState().selectedTracks[addon.key] as AddonTrack;
+
+    if (selectedTrack) {
+      return selectedTrack;
+    }
+
+    this.setCurrentSelectedTrack(addon, addon.tracks[0]);
+
+    return addon.tracks[0];
+  }
+
+  public static determineAddonInstalledTrack(addon: Addon): AddonTrack | null {
+    const installedTrack = store.getState().installedTracks[addon.key] as AddonTrack;
+
+    if (installedTrack) {
+      return installedTrack;
+    }
+
+    const install = this.getAddonInstall(Directories.inInstallLocation(addon.targetDirectory));
+
+    if (!install) {
+      return null;
+    }
+
+    const matchingTrack = addon.tracks.find((it) => it.url === install.source);
+
+    if (!matchingTrack) {
+      return null;
+    }
+
+    this.setCurrentlyInstalledTrack(addon, matchingTrack);
+    this.setCurrentSelectedTrack(addon, matchingTrack);
+
+    return matchingTrack;
+  }
+
+  private static lowestAvailableAbortControllerID(): number {
+    for (let i = 0; i < this.abortControllers.length; i++) {
+      if (
+        !store
+          .getState()
+          .downloads.map((download) => download.abortControllerID)
+          .includes(i)
+      ) {
+        return i;
+      }
+    }
+  }
+
+  private static async determineAddonInstallStatus(addon: Addon): Promise<InstallState> {
+    console.log('[InstallManager](determineAddonInstallStatus) Checking install status');
+
+    const installDir = Directories.inInstallLocation(addon.targetDirectory);
+    const addonInstalledTrack = this.determineAddonInstalledTrack(addon);
+    const addonSelectedTrack = this.getAddonSelectedTrack(addon);
+
+    if (!fs.existsSync(installDir)) {
+      console.log('[InstallManager](determineAddonInstallStatus) Is not installed');
+
+      return { status: InstallStatus.NotInstalled };
+    }
+
+    console.log('[InstallManager](determineAddonInstallStatus) Checking for git install');
+
+    if (Directories.isGitInstall(installDir)) {
+      console.log('[InstallManager](determineAddonInstallStatus) Is git install');
+
+      return { status: InstallStatus.GitInstall };
+    }
+
+    try {
+      const updateInfo = await new FragmenterUpdateChecker().needsUpdate(addonSelectedTrack.url, installDir, {
+        forceCacheBust: true,
+      });
+
+      console.log('[InstallManager](determineAddonInstallStatus) Update info', updateInfo);
+
+      if (addonSelectedTrack !== addonInstalledTrack && addonInstalledTrack) {
+        return { status: InstallStatus.TrackSwitch };
+      }
+
+      if (updateInfo.isFreshInstall) {
+        return { status: InstallStatus.NotInstalled };
+      }
+
+      if (updateInfo.needsUpdate) {
+        return { status: InstallStatus.NeedsUpdate };
+      }
+
+      return { status: InstallStatus.UpToDate };
+    } catch (e) {
+      console.error(e);
+      return { status: InstallStatus.Unknown };
+    }
+  }
+
+  public static async checkForUpdates(addon: Addon): Promise<void> {
```

Comment:
This function needs to check if the installStatus is not one of the installing/updating states before further execution due to it being called periodically in App/index.tsx

Reply:
Fixed.